x/stake/keeper/delegation.go

```diff
@@ -147,7 +168,33 @@ func (k Keeper) RemoveUnbondingDelegation(ctx sdk.Context, ubd types.UnbondingDe
 
 //_____________________________________________________________________________________
 
-// load a redelegation
+// Return all redelegations for a delegator. If maxRetrieve is supplied, the respective amount will be returned.
+func (k Keeper) GetRedelegations(ctx sdk.Context, delegator sdk.AccAddress,
+	maxRetrieve ...int16) (redelegations []types.Redelegation) {
+
+	retrieve := len(maxRetrieve) > 0
```

Comment:
we shouldn't allow for retrieve all to exist - meaning that this `retrieve` term should be taken out - see my comments on the old PR